tls-tutorial/doc/tls_in_udfs.md

```diff
@@ -0,0 +1,225 @@
+
+If you want to inject JVM arguments into a UDF, you can use the `%jvmoption` directive in the [`CREATE SCRIPT`](https://docs.exasol.com/db/latest/sql/create_script.htm) statement.
+
+```sql
+CREATE OR REPLACE JAVA SCALAR SCRIPT JAVA_TUTORIAL.CUSTOM_CERTIFICATES()
+EMITS (CN VARCHAR(2000), O VARCHAR(2000), OU VARCHAR(2000), C VARCHAR(2)) AS
+    %jvmoption -Djavax.net.ssl.trustStore=/buckets/bfsdefault/default/tls-tutorial-truststore -Djavax.net.ssl.trustStorePassword=tutorial;
+    %jar /buckets/bfsdefault/default/tls-tutorial.jar;
+    %scriptclass com.exasol.javatutorial.tls.Certificates;
+/;
+
+SELECT JAVA_TUTORIAL.CUSTOM_CERTIFICATES();
+```
+
+At this point you should see only the one entry that we added to our new trusstore earlier.
+
+| CN           | O                                | OU  | C   |
+|--------------|----------------------------------|-----|-----|
+| ISRG Root X1 | Internet Security Research Group |     | US  |
+
+### Trust Store Password and Protecting the Truststore
+
+If you followed the tutorial closely to that point, you might have wondered about the significance of the truststore password. Is it dangerous that you need to know it? Should you change the default password?
+
+Let's start by looking at what you want to protect here. IT security always falls into at least one of the following three categories: confidentiality, integrity and availability.
+
+Availability is not our main concern here, since as long as Exasol runs, BucketFS is available. And if Exasol is down, BucketFS availability is the least of your problems.
+
+By definition certificates are public information. If your protection scheme depends on keeping certificates confidential, your process is broken. That means there is no point in protecting the certificates from being read.
+
+Integrity on the other hand matters here. If an attacker manages to sneak a malicious certificate into your truststore, your certificate validation process is compromised.
+
+Does the truststore password help here? Not really. Since the same password is used for reading and writing, you cannot keep it completely secret. That would only be the case if the truststore itself used asymmetric encryption. So if the truststore password isn't the answer, you need to protect the file instead.
+
+In an operating system this is usually achieved by restricting write access to the truststore to privileged users. You can emulate this for UDFs.
+
+1. [Create a private bucket](https://docs.exasol.com/db/latest/database_concepts/bucketfs/create_new_bucket_in_bucketfs_service.htm) that only privileged users (e.g. the DBA) have write access to
+2. Chose a non-guessable write-password for the bucket
+3. Chose a non-guessable read-password for the bucket that differs from the write-password
+4. [Create a connection object for the bucket](https://docs.exasol.com/db/latest/database_concepts/bucketfs/database_access.htm) with the read-password
```

Comment:
As there are GRANT CONNECTION and GRANT ACCESS ON CONNECTION commands, we need to be careful in 4. and 5. here.
I don't see a single-word change, but anticipate some confusion.

Reply:
Agreed. Created issue #12 to fix this.